marshmallow/marshalling.py

```diff
@@ -113,9 +107,6 @@ def serialize(self, obj, fields_dict, many=False,
         .. versionchanged:: 1.0.0
             Renamed from ``marshal``.
         """
-        # Reset errors dict if not serializing a collection
-        if not self._pending:
-            self.reset_errors()
```

Comment:
> ```py
> class Marshaller(ErrorStore):
>     """Callable class responsible for serializing data and storing errors.
> ```

It seems odd to me that the `Marshaller` classes inherit from `ErrorStore`. Serializing data and storing errors seem like two separate responsibilities.